pytorch_lightning/metrics/classification/stat_scores.py

```diff
@@ -0,0 +1,261 @@
+              sample the extra axes ``...`` are flattened to become the sub-sample axis, and
+              statistics for each sample are computed by treating the sub-sample axis as the
+              ``N`` axis for that sample.
+
+            - ``'global'``: In this case the ``N`` and ``...`` dimensions of the inputs are
+              flattened into a new ``N_X`` sample axis, i.e. the inputs are treated as if they
+              were ``(N_X, C)``. From here on the ``reduce`` parameter applies as usual.
+
+        is_multiclass:
+            Used only in certain special cases, where you want to treat inputs as a different type
+            than what they appear to be (see :ref:`metrics:Input types` documentation section for
+            input classification and examples of the use of this parameter). Should be left at default
+            value (``None``) in most cases.
+
+            The special cases where this parameter should be set are:
+
+            - When you want to treat binary or multi-label inputs as multi-class or multi-dimensional
+              multi-class with 2 classes, respectively. The probabilities are interpreted as the
+              probability of the "1" class, and thresholding still applies as usual. In this case
+              the parameter should be set to ``True``.
+            - When you want to treat multi-class or multi-dimensional mulit-class inputs with 2 classes
+              as binary or multi-label inputs, respectively. This is mainly meant for the case when
+              inputs are labels, but will work if they are probabilities as well. For this case the
+              parameter should be set to ``False``.
+
+        compute_on_step:
+            Forward only calls ``update()`` and return None if this is set to False.
+        dist_sync_on_step:
+            Synchronize metric state across processes at each ``forward()``
+            before returning the value at the step. default: False
+        process_group:
+            Specify the process group on which synchronization is called. default: None (which selects the entire world)
+        dist_sync_fn:
+            Callback that performs the allgather operation on the metric state. When `None`, DDP
+            will be used to perform the allgather.
+
+    Example:
+
+        >>> from pytorch_lightning.metrics.classification import StatScores
+        >>> preds  = torch.tensor([1, 0, 2, 1])
+        >>> target = torch.tensor([1, 1, 2, 0])
+        >>> stat_scores = StatScores(reduce='macro', num_classes=3)
+        >>> stat_scores(preds, target)
+        tensor([[0, 1, 2, 1, 1],
+                [1, 1, 1, 1, 2],
+                [1, 0, 3, 0, 1]])
+        >>> stat_scores = StatScores(reduce='micro')
+        >>> stat_scores(preds, target)
+        tensor([2, 2, 6, 2, 4])
+
+    """
+
+    def __init__(
+        self,
+        threshold: float = 0.5,
+        top_k: Optional[int] = None,
+        reduce: str = "micro",
```

Comment:
to be consistent with other metrics, I would rename this arg to `average`
```suggestion
        average: str = "micro",
```

Reply:
So there's a reason I am naming this reduce:
- It's not doing any averaging, but only summing up
- StatScores will be subclassed by other metrics (Precision, Recall, ...), which will have an `average` parameter, which will have more values (in addition to micro, macro and samples also weighted and none). So either I add these possibilities here - which may look strange, as there would be no difference between macro, weighted and None, or I use a different name of the parameter